python/cuml/fil/fil.pyx

```diff
@@ -0,0 +1,219 @@
+        return self
+
+
+class FIL(Base):
+    """
+    Parameters
+    ----------
+    handle : cuml.Handle
+       If it is None, a new one is created just for this class.
+    """
+    def __init__(self,
+                 handle=None):
+        super(FIL, self).__init__(handle)
+        self._impl = FIL_impl(self.handle)
+
+    def predict(self, X, preds=None):
+        """
+        Predicts the labels for X on the model that was loaded
+        using from_treelite function.
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+           Dense matrix (floats or doubles) of shape (n_samples, n_features).
+           Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+           ndarray, cuda array interface compliant array like CuPy
+        preds: NumPy or cuDF dataframe
+           Dense vector (int) of shape (n_samples, 1)
+        Returns
+        ----------
+        labels predicted by the model
+        """
+        return self._impl.predict(X, preds)
+
+    def from_treelite(self, model, output_class, algo, threshold):
+        """
+        Creates a FIL model using the treelite model
+        passed to the function.
+        Parameters
+        ----------
+        model : the model information in the treelite format
+           loaded from a saved model using the treelite API
+           https://treelite.readthedocs.io/en/latest/treelite-api.html
+        output_class: boolean
+           True or False
+        algo : 0 = NAIVE, 1 = TREE_REORG, 2 = BATCH_TREE_REORG
+        threshold : threshold is used to for classification
+           if output == OUTPUT_CLASS, else it is ignored
+        """
+        return self._impl.from_treelite(model, output_class, algo, threshold)
+
+
+def from_treelite_direct(model, algo=0, output_class=True,
+                         threshold=0.5, handle=None):
+    """
+    Creates a FIL model using the treelite model
+    passed to the function.
+    Note : from cuml import fil
+           forest = fil.from_treelite_direct(params)
+           # in order to predict
+           predicted_labels = forest.predict(X_test)
+
+    Note : do not reuse or overwrite the tl_model variable which is
+       used to load a saved model.
+       This will cause Segmentation errors.
+       here tl_model = = treelite.Model.load('xgb.modle', 'xgboost')
+
+    Parameters
+    ----------
+    model : the model information in the treelite format
+       loaded from a saved model using the treelite API
+       https://treelite.readthedocs.io/en/latest/treelite-api.html
+    output_class: boolean
+       True or False
+    algo : 0 = NAIVE, 1 = TREE_REORG, 2 = BATCH_TREE_REORG
+    threshold : threshold is used to for classification if
+       output == OUTPUT_CLASS, else it is ignored
+    handle : cuml.Handle
+       If it is None, a new one is created just for this class.
+    """
+    fil_model = FIL(algo=algo, threshold=threshold, handle=handle)
```

Comment:
The docs say we will create a cumlHandle as needed... where does that happen? I don't see it at the lower layer

Reply:
` super(FIL, self).__init__(handle)` is used to create the handle if it is none